libexif/exif-data.c

```diff
@@ -1044,6 +1044,30 @@ exif_data_new_from_file (const char *path)
 
 	return (edata);
 }
+ExifData *
+exif_data_new_from_fd (int fd, const char* path, unsigned int *bytes_read) {
```

Comment:
What's the use case for this? Applications can already load data from a file or from memory buffers. I don't see a compelling need for this. Also, path as an argument isn't necessary.